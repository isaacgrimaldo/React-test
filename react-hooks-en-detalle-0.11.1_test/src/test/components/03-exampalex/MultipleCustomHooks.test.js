import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

import {MultipleCustomHooks} from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Puebas del componente MultipleCustomHooks', () => {


     test('Debe de mostarse correctamente el componete', () => {

        useCounter.mockReturnValue({
            counter: 10, 
            increment : () => {},
           })

        useFetch.mockReturnValue({ 
           data:null, 
           loading:true,
           error:null       
        })

        const wrapper = shallow( <MultipleCustomHooks/> );
        expect(wrapper).toMatchSnapshot();
     });

     test('debe de mostrar informacion resivada', () => {
       useFetch.mockReturnValue({
           data:[
               {
                   author:'Isaac', 
                   quote:'Hola Mundo'
               }  
          ],
           loading:false,
           error:null,
       });

       const wrapper = shallow( <MultipleCustomHooks/> );
       expect(wrapper.find('.alert').exists()).toBe(false);
       expect(wrapper.find('.mb-0').text().trim()).toBe('Hola Mundo');
     });
});