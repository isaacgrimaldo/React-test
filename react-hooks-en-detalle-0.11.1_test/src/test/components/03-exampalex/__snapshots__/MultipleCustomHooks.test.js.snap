// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas del componente MultipleCustomHooks Debe de mostarse correctamente el componete 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
